pkg/virt-controller/watch/vm.go

```diff
@@ -3135,31 +3141,17 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 
 	vmCopy := vm.DeepCopy()
 
-	if c.clusterConfig.HotplugNetworkInterfacesEnabled() &&
-		vmi != nil && vmi.DeletionTimestamp == nil {
-		vmiCopy := vmi.DeepCopy()
-
-		indexedStatusIfaces := vmispec.IndexInterfaceStatusByName(vmi.Status.Interfaces,
-			func(ifaceStatus virtv1.VirtualMachineInstanceNetworkInterface) bool { return true })
-
-		ifaces, networks := network.ClearDetachedInterfaces(vmCopy.Spec.Template.Spec.Domain.Devices.Interfaces, vmCopy.Spec.Template.Spec.Networks, indexedStatusIfaces)
-		vmCopy.Spec.Template.Spec.Domain.Devices.Interfaces = ifaces
-		vmCopy.Spec.Template.Spec.Networks = networks
-
-		ifaces, networks = network.ClearDetachedInterfaces(vmiCopy.Spec.Domain.Devices.Interfaces, vmiCopy.Spec.Networks, indexedStatusIfaces)
-		vmiCopy.Spec.Domain.Devices.Interfaces = ifaces
-		vmiCopy.Spec.Networks = networks
-
-		hasOrdinalIfaces, err := c.hasOrdinalNetworkInterfaces(vmi)
-		if err != nil {
-			return vm, &syncErrorImpl{fmt.Errorf("Error encountered when trying to check if VMI has interface with ordinal names (e.g.: eth1, eth2..): %v", err), HotPlugNetworkInterfaceErrorReason}, nil
-		}
-		updatedVmiSpec := network.ApplyDynamicIfaceRequestOnVMI(vmCopy, vmiCopy, hasOrdinalIfaces)
-		vmiCopy.Spec = *updatedVmiSpec
-
-		if err := c.vmiInterfacesPatch(&vmiCopy.Spec, vmi); err != nil {
-			return vm, &syncErrorImpl{fmt.Errorf("Error encountered when trying to patch vmi: %v", err), FailedUpdateErrorReason}, nil
+	for _, syncer := range c.Synchronizers {
+		resVM, errSync := syncer.Sync(vmCopy, vmi)
```

Comment:
Not sure what `resVM` means, please consider renaming it to something like syncedVM/updatedVM.

nit: I would rename `errSync` to plain `err`.

Reply:
> Not sure what `resVM` means, please consider renaming it to something like syncedVM/updatedVM.

Done
 
> nit: I would rename `errSync` to plain `err`.

I prefer not to shadow the upper `err`.